tools/run-fbp-tests

```diff
@@ -76,137 +174,130 @@ def collect_tests(tests):
             result.append(t)
     return result
 
-def get_expected(t, args):
-    code = 0
-    out = None
-    skip = None
-    is_regex = False
+def valgrind_prepare(args):
+    cmd_prefix = []
 
-    f = open(t, encoding="UTF-8")
-    for line in f:
-        if not line.startswith("## "):
-            continue
-        splitted = line.split(None, 2)
-        cmd = splitted[1]
+    if not args.valgrind:
+        return cmd_prefix
 
-        if cmd == "TEST-OUTPUT-REGEX":
-            is_regex = True
-
-        if cmd == "TEST-EXPECTS-ERROR":
-            code = 1
-        elif cmd == "TEST-SKIP-VALGRIND":
-            if args.valgrind:
-                if len(splitted) > 2:
-                    skip = splitted[2]
-                else:
-                    skip = ""
-        elif cmd == "TEST-SKIP":
-            if len(splitted) > 2:
-                skip = splitted[2]
-            else:
-                skip = ""
-        elif cmd in ["TEST-OUTPUT", "TEST-OUTPUT-REGEX"]:
-            out = ""
-            break
-        else:
-            print("Ignoring unknown test command '%s' in test %s" % (cmd, t))
-            continue
+    valgrind = os.path.abspath(args.valgrind)
+    if not os.path.isfile(valgrind):
+        sys.exit("Couldn't find valgrind program at", valgrind)
```

Comment:
Might be a good idea to see if the file is executable rather than just check if it exists.